### PR TITLE
Add search address and detail address nodes to portfolio graph visualization

### PR DESCRIPTION
**Pull request by sraby**
This PR refactors our `<PortfolioGraph>` component to dynamically add on nodes to the graph visualization for the user's Search Address, as well as the Detail Address if it differs from the Search Address. The idea behind these additional nodes was to create some feature parity with the "How is this building associated..." feature on the old WOW:

image

Now, when users click on this link, they are redirected to the Portfolio Graph (currently on the Summary tab) to see how their search address is related to the address in focus on the Overview tab:

image

This is particularly important because in order to build trust with our users regarding this new portfolio method, we need to have some feature that illustrates how each building in the portfolio connects back to the original address they searched.

Potential further improvements/considerations:
- adding better labels to the two additional nodes on the graph (like `Search address: ...` and `Focus address: ...`) maybe?
- updating the colors/style to pop out a bit more
- making sure that the graph is fully "unfurled" with the 2 new nodes added (little to no overlapping edges)

### CODE REVIEW

**Comment by sraby** [comment]
[ch7985]

File: client/src/components/DetailView.tsx [hunk 1]
```diff
@@ -207,15 +212,22 @@ class DetailViewWithoutI18n extends Component<Props, State> {
...
+                            <i>
+                              <BldgAssociationHeader />
+                            </i>
+                          </Link>
+                        ) : (
+                          <a // eslint-disable-line jsx-a11y/anchor-is-valid
+                            onClick={() => this.setState({ showCompareModal: true })}
+                          >
+                            <i>
+                              <BldgAssociationHeader />
```
Comment (samaratrilling) [inline client/src/components/DetailView.tsx:227]: This sounds like a tenant association within a building! lol i got confused. Maybe PortfolioAssociationHeader?

File: client/src/components/PortfolioGraph.tsx [hunk 2]
```diff
@@ -3,55 +3,163 @@ import Cytoscape from "cytoscape";
...
+  }
+  return additionalNodes;
+}
+
+/**
+ * Given an existing array of graph nodes and an Address Record, this function
+ * finds the `id` property of a node of type `name` that corresponds to the owner
+ * (or owners) listed with the address.
+ */
+export function getLandlordNodeIDFromAddress(
```
Comment (samaratrilling) [inline client/src/components/PortfolioGraph.tsx:75]: any chance you got to do some profiling of this on larger sets? how does the performance seem?

**Review by samaratrilling — APPROVED** [review]

File: client/src/components/PortfolioGraph.tsx (hunk 2 above)
Comment (sraby) [reply client/src/components/PortfolioGraph.tsx:75]: lemme check!

File: client/src/components/DetailView.tsx (hunk 1 above)
Comment (sraby) [reply client/src/components/DetailView.tsx]: Ok did a quick profile and it seems like the computation is pretty instant even for our largest portfolio! Check out this cute pic too and see if u can spot our new bbls: 
image

File: client/src/components/PortfolioGraph.tsx (hunk 2 above)
Comment (sraby) [reply client/src/components/PortfolioGraph.tsx:75]: Ok did a quick profile and it seems like the computation is pretty instant even for our largest portfolio! Check out this cute pic too and see if u can spot our new bbls: 
image

Comment (samaratrilling) [reply client/src/components/PortfolioGraph.tsx:75]: 🤩 bb's first bbls 🤩

**Pull request merged**